diff -urN stlink.ORIG/CMakeLists.txt stlink/CMakeLists.txt
--- stlink.ORIG/CMakeLists.txt	2020-02-27 01:56:59.000000000 +0100
+++ stlink/CMakeLists.txt	2020-03-13 09:04:49.200305243 +0100
@@ -185,18 +185,10 @@
 # Tools
 ###
 add_executable(st-flash src/tools/flash.c src/tools/flash_opts.c)
-if (WIN32 OR APPLE)
-	target_link_libraries(st-flash ${STLINK_LIB_STATIC})
-else()
-	target_link_libraries(st-flash ${STLINK_LIB_SHARED})
-endif()
+target_link_libraries(st-flash ${STLINK_LIB_STATIC})
 
 add_executable(st-info src/tools/info.c)
-if (WIN32 OR APPLE)
-	target_link_libraries(st-info ${STLINK_LIB_STATIC})
-else()
-	target_link_libraries(st-info ${STLINK_LIB_SHARED})
-endif()
+target_link_libraries(st-info ${STLINK_LIB_STATIC})
 
 install(TARGETS st-flash st-info
 	RUNTIME DESTINATION bin
diff -urN stlink.ORIG/src/usb.c stlink/src/usb.c
--- stlink.ORIG/src/usb.c	2020-02-27 01:56:59.000000000 +0100
+++ stlink/src/usb.c	2020-03-16 12:55:28.750739499 +0100
@@ -189,7 +189,7 @@
         cmd[i++] = STLINK_APIV3_GET_VERSION_EX;
 
         size = send_recv(slu, 1, cmd, slu->cmd_len, data, rep_len);
-        if (size != rep_len) {
+        if (size != (ssize_t)rep_len) {
             printf("[!] send_recv STLINK_APIV3_GET_VERSION_EX\n");
             return (int) size;
         }
